python/paddle/fluid/incubate/fleet/base/fleet_base.py

```diff
@@ -189,18 +189,8 @@ def init(self, role_maker=None):
         if role_maker and not isinstance(role_maker, RoleMakerBase):
             raise ValueError("role_maker must be an instance of RoleMakerBase")
 
-        if isinstance(role_maker, MPISymetricRoleMaker):
-            self._role_maker = role_maker
-            self._role_maker.generate_role()
-
-        elif isinstance(role_maker, UserDefinedRoleMaker):
-            self._role_maker = role_maker
-
-        else:
-            raise ValueError(
-                "role_maker must be an instance of UserDefinedRoleMaker/MPISymetricRoleMaker"
-            )
-
+        self._role_maker = role_maker
+        self._role_maker.generate_role()
```

Comment:
Not all RoleMaker have the method 'generate_role()'

Reply:
generate_role() is base class method that can be called. default implementation is do nothing.